### ISSUE TITLE
NPC clipping issues in Tiragarde Keep, Durotar

### ISSUE THREAD

**Issue by eTzmNcbkrng** [chromiecraft/chromiecraft#128]
##### ISSUE: FACTION SIDE:

Probably only affects Horde, but unsure.

##### CONTENT PHASE:

1-19

##### SMALL DESCRIPTION:

Quest [Vanquish the Betrayers](https://www.wowhead.com/quest=784/) requires you to go to Tiragarde Keep in Durotar
There are two main NPCs in Tiragarde Keep, [Kul Tiras Sailor](https://www.wowhead.com/npc=3128/)s and [Kul Tiras Marine](https://www.wowhead.com/npc=3129/)s. Both of these NPCs flee when they get low health, when they flee, they're able to run through the stone walls/fences around the keep and are also able to run through the walls of the keep (from inside to outside and from outside to inside).

##### EXPECTED BLIZZLIKE BEHAVIOUR:

NPCs should not be able to walk through walls

##### CURRENT BEHAVIOUR:

NPCs can walk through walls in Tiragarde Keep area

##### STEPS TO REPRODUCE THE PROBLEM:

Step 1 Attack Kul Tiras Marine or Kul Tiras Sailor
Step 2 Do damage until NPC is low health and it flees
Step 3 NPC will sometimes flee through a wall

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/c2f274e06dcf85c6ccc0ab282dd6f23e8ce49c8f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by bonfax4** [comment]
It's not just those npc's. Every fleeing mob inside or around structures seem to suffer from this.I've had npc's die inside walls as well with dots. I do wonder if this is related to the same issue that causes warlock and priest fear to ignore collision as well.

**Comment by Stifler82** [comment]
So guys, I'll close this report, but before I close it it's important that I explain why I'm closing it.
This is a known problem to the Staff, because it concerns the Vmaps system.

Now I will explain how it works by bringing comparisons with real life to make the concept "more digestible".

the Vmaps manage some concepts of game physics, as in the real world there are different types of textures such as metal, wood, stone, etc, etc.. so, physic's laws, also in Game there are different types of material consistencies and different physic's laws.
Now, Vmaps handle these consistencies, so it's not a problem of:  Quest, NPC, or whatever...

Basically, all problems related to:
Npc going through walls/structures, all the "Fear Spells-like" that send you into the wall, and all those problems related to game physics, are caused by Vmaps, and we are already working on this, but it is not very simple thing to fix, so it would be very welcome and appreciated by us (Staff) if you could discern between a Physics Bug and a Subjective Bug of an NPC/Quest/GameObject/etc..

And so, if you guys avoided posting this kind of Bugs, because that's something we already know, and we're already working on it.

**Comment by Stifler82** [comment]
https://github.com/chromiecraft/chromiecraft/issues/303

**Comment by Stifler82** [comment]
@bonfax4 @eTzmNcbkrng 
Now it seems work well.
**Video Test:** [Flee for Assist](https://youtu.be/VPUW2eXQric)
